pkg/proxy/iptables/proxier.go

```diff
@@ -788,11 +791,25 @@ func (proxier *Proxier) execConntrackTool(parameters ...string) error {
 // assumes proxier.mu is held
 func (proxier *Proxier) syncProxyRules() {
 	if proxier.throttle != nil {
-		proxier.throttle.Accept()
+		// NOTE: using non-blocking TryAccept.
+		// All other updates should update the datastructure
+		// and do the appropriate thing to batch when we are not syncing.
+		// Lastly, the sync timer will be set to ensure we catch the last update.
+		if proxier.throttle.TryAccept() == false {
+			duration := time.Since(proxier.lastSync)
+			if duration < proxier.minSyncPeriod {
```

Comment:
@timothysc The #36006 patch checks for no-op after any rate limits would be applied though, so that won't help this issue.